rts/src/main/java/eta/runtime/io/MemoryManager.java

```diff
@@ -237,4 +242,161 @@ public static long loadStringsUTF8(String[] ss) throws UnsupportedEncodingExcept
     {
         return loadStrings(ss, "UTF-8");
     }
+
+    public static ByteBuffer getBoundedBuffer(long ptr, int offset, int length) {
+        ByteBuffer buf = getBoundedBuffer(ptr);
+        buf.position(buf.position() + offset);
+        buf.limit(buf.position() + length);
+        return buf;
+    }
+
+    public static byte[] getBytes(long ptr, int offset, int length) {
+        ByteBuffer buf = getBoundedBuffer(ptr,offset,length);
+        byte[] b = new byte[buf.remaining()];
+        buf.get(b);
+        return b;
+    }
+    
+    public static ByteBuffer copyByteBuffer( ByteBuffer src, ByteBuffer dest, int n) {
+        return copyByteBuffer(src, 0, dest, 0, n);
+    }
+
+    public static ByteBuffer copyByteBufferDups( ByteBuffer src, ByteBuffer dest, int n) {
+        return copyByteBufferDups(src, 0, dest, 0, n);
+    }
+    
+    public static ByteBuffer copyByteBuffer( ByteBuffer src, int srcOffset
+                                           , ByteBuffer dest, int destOffset
+                                           , int n) {
+        if (src == dest)
```

Comment:
Will this yield the correct behavior when `srcOffset != destOffset`?

Reply:
There is [a test for that case](https://github.com/typelead/eta/pull/629/files/491c564a4a756fb29b9c430c567e49669d198e43#diff-3a696de3cc773184d32d7b592119ff16R39), although maybe it is not correct...